### PR TITLE
test: make tests consistent on machines which have different timezone…

### LINKED ISSUE #6931: test: "database schema > column types > postgres" is not working with different timezone
**Issue type:**

[x] bug report
[ ] feature request
[x] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[x] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

1. Open your postgresql.conf
2. Set:

```conf
timezone = 'Europe/Berlin'
```

Screen Shot 2020-10-18 at 14 19 25

This tests fails if you have a different timezone for your local postgres instance.

### PR DESCRIPTION
**Pull request by TheNoim**
…s configured for postgres

With this change, you can run the test local without changing your timezone in the postgres config. Before this change, two tests would fail, because of a different timezone in postgres

Closes #6931

### Description of change

When I was working on #6469 I noticed I can't run all test local on my machine, because I had a different timezone configured for my postgres test instance. This is why I always only run the test relevant for me on my local machine and test the rest of the test I only run as part of the CI process. With this change, I am able to run all test on my local machine. 

### Pull-Request Checklist

- [X] Code is up-to-date with the `master` branch
- [x] `npm run lint` passes with this change
- [X] `npm run test` passes with this change
- [X] This pull request links relevant issues as `Fixes #0000`
- [X] There are new or updated unit tests validating the change
- [ ] Documentation has been updated to reflect this change
- [X] The new commits follow conventions explained in [CONTRIBUTING.md](https://github.com/typeorm/typeorm/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: test/functional/database-schema/column-types/postgres/column-types-postgres.ts [hunk 1]
```diff
@@ -4,18 +4,38 @@ import {Connection} from "../../../../../src/connection/Connection";
 import {closeTestingConnections, createTestingConnections, reloadTestingDatabases} from "../../../../utils/test-utils";
 import {PostWithoutTypes} from "./entity/PostWithoutTypes";
 import {Post} from "./entity/Post";
+import {PostgresDriver} from "../../../../../src/driver/postgres/PostgresDriver";
 
 describe("database schema > column types > postgres", () => {
+    // Save the original set timezone for later restore
+    let originalTimezone: string;
```
Comment (pleerock) [inline test/functional/database-schema/column-types/postgres/column-types-postgres.ts:11]: do we really need to "restore it"? Isn't `SET TIME ZONE` per-session configuration?

Comment (TheNoim) [reply test/functional/database-schema/column-types/postgres/column-types-postgres.ts:11]: I thought it is set until restart of postgres

**Comment by pleerock** [comment]
What if we make this connection-configuration-level option? (e.g. `timezone` option ormconfig / ConnectionOptions)

I think it's better to keep UTC in all places, and where specific timezone is needed, we can manually trigger SET TIME ZONE within `query` method.

**Comment by TheNoim** [comment]
> What if we make this connection-configuration-level option? (e.g. `timezone` option ormconfig / ConnectionOptions)
> 
> I think it's better to keep UTC in all places, and where specific timezone is needed, we can manually trigger SET TIME ZONE within `query` method.

I think this is a breaking change, or isn't it? I mean, this change here only effects tests and make them work on a different postgres configuration. It is probably a hint for a deeper issue.

**Comment by pleerock** [comment]
> I think this is a breaking change, or isn't it?

no, it's optional property and can be set by user. I think it can be helpful for some users.

> I mean, this change here only effects tests and make them work on a different postgres configuration.

right but I think users can have same issues.

Maybe if you can checkout exist issues you can find something related.

**Comment by TheNoim** [comment]
> > I think this is a breaking change, or isn't it?
> 
> no, it's optional property and can be set by user. I think it can be helpful for some users.
> 
> > I mean, this change here only effects tests and make them work on a different postgres configuration.
> 
> right but I think users can have same issues.
> 
> Maybe if you can checkout exist issues you can find something related.

Hey, let's do a deal. This fixes the issue at least for testing right now. How about you merge it (if I remove this timezone restore if necessary) and I try to find the underlining issue later. If I fix it, I can revert the changes in the tests again. But at least it wouldn't be an issue for local testing anymore.

**Comment by pleerock** [comment]
> Hey, let's do a deal. This fixes the issue at least for testing right now. How about you merge it (if I remove this timezone restore if necessary) and I try to find the underlining issue later. If I fix it, I can revert the changes in the tests again. But at least it wouldn't be an issue for local testing anymore.

right, it's better to have it rather not to have it. But you probably know nothing is more permanent than temporary. I strongly suggest to make it the way I proposed. If you disagree - I'll merge it.

File: test/functional/database-schema/column-types/postgres/column-types-postgres.ts (hunk 1 above)
Comment (TheNoim) [reply test/functional/database-schema/column-types/postgres/column-types-postgres.ts]: It is unnecessary. I removed it.

**Pull request merged**